File: src/main/java/yace/Application.java
package yace;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.Optional;

public record Application(Path source, Path target) {
    Optional<Path> run() throws IOException {
        if (Files.exists(source())) {
            Files.createFile(target());
            return Optional.of(target());
        }
        return Optional.empty();
    }
}

File: src/test/kotlin/Application.kt
import java.nio.file.Files
import java.nio.file.Path

class Application(private val source: Path, private val gateway: PathGateway) {
    fun run(): Path? {
        if (Files.exists(source)) {
            val nameCount = source.nameCount
            val lastIndex = nameCount - 1
            val lastValueWithSeparator = source.getName(lastIndex).toString()
            val separator = lastValueWithSeparator.indexOf('.')
            val lastValue = lastValueWithSeparator.substring(0, separator)
            val target = gateway.resolveTarget(lastValue)
            Files.createFile(target)
            return target
        }
        return null
    }
}